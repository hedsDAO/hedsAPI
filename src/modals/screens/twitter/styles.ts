import { ModalProps, ModalContentProps, ModalHeaderProps, ModalCloseButtonProps } from '@chakra-ui/react';

export const $twitterModalStyles: Partial<ModalProps> = {
  size: 'lg',
  motionPreset: 'slideInBottom',
  isCentered: true,
};

export const $twitterModalContentStyles: ModalContentProps = {
  rounded: 'xl',
  m: 2,
};

export const $twitterModalHeaderStyles: ModalHeaderProps = {
  fontWeight: 'bold',
  color: 'heds.200',
  fontSize: 'md',
  letterSpacing: 'widest',
};

export const $twitterModalCloseButtonStyles: ModalCloseButtonProps = {
  _focus: { boxShadow: 'none' },
  color: 'white',
};
