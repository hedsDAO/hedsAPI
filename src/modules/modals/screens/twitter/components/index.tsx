export { default as CopyTweetForm } from './CopyTweetForm/CopyTweetForm';
export { default as GenerateHashForm } from './GenerateHashForm/GenerateHashForm';
export { default as PasteTweetForm } from './PasteTweetForm/PasteTweetForm';
export { default as TweetHashForm } from './TweetHashForm/TweetHashForm';
export { default as VerifyAndLinkAccountForm } from './VerifyAndLinkAccountForm/VerifyAndLinkAccountForm';
