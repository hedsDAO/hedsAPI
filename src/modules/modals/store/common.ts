/**
 * @name SampleModal
 * @summary types, text and data used for the sample download modal.
 *
 */

export class SampleModalText {
  title = 'Download Sample';
  icon = 'fa-light fa-waveform-lines mr-1 text-xl';
  h1 = 'Requirements';
  req1 = 'bpm:';
  req2 = 'duration:';
  primaryAlertText = 'Public submissions end in:';
  warningAlertText = 'Public submissions have ended.';
  checkboxText = 'I understand the submission requirements.';
  primaryButtonText = 'Download';
  secondaryButtonText = 'Back';
}

/**
 * @name SubmitModal
 * @summary types, text and data used for the sample download modal.
 *
 */

export class SubmitModal {
  title = '';
  icon = '';
  h1 = '';
  req1 = '';
  req2 = '';
  primaryAlertText = '';
  warningAlertText = '';
  checkboxText = '';
  primaryButtonText = '';
  secondaryButtonText = '';
}
