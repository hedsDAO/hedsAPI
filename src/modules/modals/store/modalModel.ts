import { createModel } from '@rematch/core';
import type { RootModel } from '@/models';

export enum Modals {
  SETTINGS_MODAL = 0,
  TWITTER_MODAL,
  NAME_MODAL,
  SAMPLE_MODAL,
  SUBMIT_MODAL,
  MINT_MODAL,
}

export interface ModalState {
  currentModal: Modals;
  nextModal?: Modals;
  isOpen: boolean;
}

export const modalModel = createModel<RootModel>()({
  state: {
    isOpen: false,
  } as ModalState,
  reducers: {
    setModal: (state, currentModal) => ({ ...state, currentModal }),
    setModalOpen: (state, isOpen: boolean) => ({ ...state, isOpen }),
    setNextModal: (state, nextModal: Modals) => ({ ...state, nextModal }),
  },
  effects: () => ({}),
});
