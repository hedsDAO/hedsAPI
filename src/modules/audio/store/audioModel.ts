import { createModel } from '@rematch/core';
import type { RootModel } from '../../../models';
import { TapeData, TrackMetadata, User } from '../../../models/common';

export interface AudioState {
  tapes: Array<TapeData>;
  queue: Array<TrackMetadata>;
  artists: Array<User>;
  currentTrack: number;
  currentTapeId: any;
  currentTape: string;
  isPlaying: boolean;
  isLoading: boolean;
  isShowingQueue: boolean;
  volume: number;
  isSample: boolean;
  currentTime: [string, number];
  duration: [string, number];
}

export const audioModel = createModel<RootModel>()({
  state: {
    isPlaying: false,
    isLoading: false,
    isShowingQueue: false,
    volume: 100,
    queue: [],
    tapes: [],
    artists: [],
  } as AudioState,
  reducers: {
    pushToQueue: (state, [track, tape, artist]: [TrackMetadata, TapeData, User]) => {
      const newState: AudioState = { ...state };
      const queue = [...newState.queue, track];
      const tapes = [...newState.tapes, tape];
      const artists = [...state.artists, artist];
      return { ...state, queue, tapes, artists };
    },
    setIsShowingQueue: (state, isShowingQueue: boolean) => ({...state, isShowingQueue}),
    setIsPlaying: (state, isPlaying: boolean) => ({ ...state, isPlaying }),
    setVolume: (state, volume: number) => ({ ...state, volume }),
    setCurrentTime: (state, currentTime: [string, number]) => ({ ...state, currentTime }),
    setDuration: (state, duration: [string, number]) => ({ ...state, duration }),
    setIsLoading: (state, isLoading: boolean) => ({ ...state, isLoading }),
  },
  effects: () => ({}),
});
