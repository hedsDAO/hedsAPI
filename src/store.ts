import { init, RematchDispatch, RematchRootState } from '@rematch/core';
import loadingPlugin, { ExtraModelsFromLoading } from '@rematch/loading';
import persistPlugin from '@rematch/persist';
import storage from 'redux-persist/lib/storage';
import selectPlugin from '@rematch/select';
import { models, RootModel } from './models';
import { createTransform } from 'redux-persist';
import { AudioModelState } from './hooks/useAudio/models/common';

// Custom transform for isPlaying property
const setIsPlayingFalseTransform = createTransform(
  // Outbound (from state to storage)
  (outboundState: unknown, key) => {
    if (key === 'audioModel') {
      const castedState = outboundState as AudioModelState;
      return {
        ...castedState,
        isPlaying: false,
      };
    }
    return outboundState;
  },
  // Inbound (from storage to state)
  (inboundState: unknown, key) => {
    if (key === 'audioModel') {
      const castedState = inboundState as AudioModelState;
      return {
        ...castedState,
        isPlaying: false,
      };
    }
    return inboundState;
  },
);

type FullModel = ExtraModelsFromLoading<RootModel>;

/** Local Storage Persist for Audio Model */
const persistConfig = {
  key: 'root',
  storage,
  version: 2,
  whitelist: ['audioModel'],
  transforms: [setIsPlayingFalseTransform],
};

export const store = init<RootModel, FullModel>({
  models,
  plugins: [selectPlugin(), loadingPlugin(), persistPlugin(persistConfig)],
  redux: {
    devtoolOptions: {
      disabled: process.env.NODE_ENV !== 'development',
    },
  },
});

export type Store = typeof store;
export type Dispatch = RematchDispatch<RootModel>;
export type RootState = RematchRootState<RootModel, FullModel>;
