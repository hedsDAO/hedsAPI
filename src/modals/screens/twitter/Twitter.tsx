import { useSelector } from 'react-redux';
import { GenerateAndCopyTweet } from '@/modals/screens/twitter/components/GenerateAndCopyTweet/GenerateAndCopyTweet';
import { useTwitter } from '@/modals/screens/twitter/components/useTwitter/useTwitter';
import { VerifyTweet } from '@/modals/screens/twitter/components/VerifyTweet/VerifyTweet';
import { Confirm } from '@/modals/screens/twitter/components/Confirm/Confirm';
import { TwitterModalSteps } from '@/modals/screens/twitter/models/common';
import { TWITTER_MODAL_HEADING } from '@/modals/screens/twitter/models/constants';
import { store } from '@/store';
import { Modal, ModalBody, ModalCloseButton, ModalContent, ModalHeader, ModalOverlay, Text } from '@chakra-ui/react';

export const Twitter = () => {
  const { isOpen, handleClose } = useTwitter();
  const currentStep = useSelector(store.select.twitterModel.selectCurrentStep);
  return (
    <Modal size="lg" motionPreset="slideInBottom" isCentered isOpen={isOpen} onClose={handleClose}>
      <ModalOverlay />
      <ModalContent rounded="xl" m={2}>
        <ModalHeader>
          <Text fontWeight={'bold'} color="heds.200" fontSize={'md'} letterSpacing="widest">
            {TWITTER_MODAL_HEADING}
          </Text>
        </ModalHeader>
        <ModalCloseButton color="white" />
        <ModalBody>
          {currentStep === TwitterModalSteps.GENERATE_AND_COPY_TWEET && <GenerateAndCopyTweet />}
          {currentStep === TwitterModalSteps.VERIFY_TWEET && <VerifyTweet />}
          {currentStep === TwitterModalSteps.CONFIRM && <Confirm />}
        </ModalBody>
      </ModalContent>
    </Modal>
  );
};
