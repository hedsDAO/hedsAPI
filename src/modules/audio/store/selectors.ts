import { TrackMetadata } from '@/models/common';
import { RootState } from '@/store';

export const selectActiveTrack = (state: RootState) => state.audioModel?.activeTrack;
export const selectQueue = (state: RootState) => state.audioModel?.queue;
export const selectIsLoading= (state: RootState) => state.audioModel?.isLoading;
export const selectIsShowingPlayer = (state: RootState) => state.audioModel?.isShowingPlayer;
export const selectIsShowingQueue = (state: RootState) => state.audioModel?.isShowingQueue;
export const selectIsQueueEmpty = (state: RootState) => state.audioModel?.queue?.length ? false : true;
export const selectAudioVolume = (state: RootState) => state.audioModel?.volume;
export const selectIsTrackPlaying = (state: RootState) => state.audioModel?.isPlaying;
export const selectTimerSeconds = (state: RootState) => state.audioModel?.timerSeconds;
export const selectCountPlayThreshold = (state: RootState) => state.audioModel?.countPlayThreshold;
export const selectActiveTrackStats = (state: RootState) => state.audioModel?.activeTrack?.stats;