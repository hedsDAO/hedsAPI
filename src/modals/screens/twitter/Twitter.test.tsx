import { BrowserRouter as Router } from 'react-router-dom';
import { Twitter } from '@/modals/screens/twitter/Twitter';
import { store } from '@/store';
import { renderWithRematchStore } from '@/tests/renderWithRematchStore';

describe('Twitter unit', () => {
  beforeAll(async () => {});
  describe('snapshot', () => {
    test('snapshot', async () => {
      const snapshot = renderWithRematchStore(
        <Router>
          <Twitter />
        </Router>,
        store,
      );
      expect(snapshot).toMatchSnapshot();
    });
  });
});
