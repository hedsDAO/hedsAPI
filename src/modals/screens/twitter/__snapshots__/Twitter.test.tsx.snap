// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Twitter unit snapshot snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
    />
    <div
      class="chakra-portal"
    >
      <div
        aria-hidden="true"
        class="chakra-modal__overlay css-13d1tkw"
        data-aria-hidden="true"
        style="opacity: 0;"
      />
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="0"
      />
      <div
        data-focus-lock-disabled="false"
      >
        <div
          class="chakra-modal__content-container css-1oxhx2p"
          tabindex="-1"
        >
          <section
            aria-describedby="chakra-modal--body-:r1:"
            aria-labelledby="chakra-modal--header-:r1:"
            aria-modal="true"
            class="chakra-modal__content css-xikv3d"
            id="chakra-modal-:r1:"
            role="dialog"
            style="opacity: 0; transform: translateX(0px) translateY(16px) translateZ(0);"
            tabindex="-1"
          >
            <header
              class="chakra-modal__header css-72fd9l"
              id="chakra-modal--header-:r1:"
            >
              <p
                class="chakra-text css-oh71vy"
              >
                VERIFICATION
              </p>
            </header>
            <button
              aria-label="Close"
              class="chakra-modal__close-btn css-1ylqxb2"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <div
              class="chakra-modal__body css-0"
              id="chakra-modal--body-:r1:"
            >
              <div
                class="chakra-stack css-chdadr"
              >
                <p
                  class="chakra-text css-1s2oizo"
                >
                  Generate your tweet
                </p>
                <p
                  class="chakra-text css-10jjrvo"
                >
                  Generate a unique tweet to help us connect your twitter to your account
                </p>
                <div
                  class="css-1hontid"
                >
                  <input
                    aria-readonly="true"
                    class="chakra-input css-5wizkk"
                    data-testid="twitter-input"
                    disabled=""
                    readonly=""
                    value=""
                  />
                  <button
                    class="chakra-button css-112drys"
                    type="button"
                  >
                    GENERATE
                  </button>
                  <button
                    class="chakra-button css-15mufsz"
                    disabled=""
                    type="button"
                  >
                    NEXT
                  </button>
                </div>
              </div>
            </div>
          </section>
        </div>
      </div>
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
    data-aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
